src/cmd_line/commands/normal.ts

```diff
@@ -0,0 +1,35 @@
+import { Parser, all, seq, whitespace } from 'parsimmon';
+import { VimState } from '../../state/vimState';
+import { ExCommand } from '../../vimscript/exCommand';
+import { LineRange } from '../../vimscript/lineRange';
+
+export class NormalCommand extends ExCommand {
+  public static readonly argParser: Parser<NormalCommand> = seq(whitespace, all).map(
+    ([_, keystroke]) => new NormalCommand(keystroke),
+  );
+
+  private readonly keystroke: string;
+  constructor(argument: string) {
+    super();
+    this.keystroke = argument;
+  }
+
+  override async execute(vimState: VimState): Promise<void> {
+    const keystroke = this.keystroke;
+    vimState.recordedState.transformer.addTransformation({
```

Comment:
I think that not setting start/end is better, because normal command behavior with visual and normal mode are different.
for example, 
- text: `aaaaa`
- start position: end of line
- command: `:normal iX`

executing above with visual mode, the result is `Xaaaaa`.
but with normal mode, it is `aaaaaX`.

to achive the difference, `executeTransformations` checks setting start/end or not.
`withRange` stores it as boolean.
```
if (withRange) {
vimState.cursorStopPosition = vimState.cursorStartPosition =
    TextEditor.getFirstNonWhitespaceCharOnLine(vimState.document, line.lineNumber);
}
await modeHandler.handleMultipleKeyEvents(keystroke.value);
await vimState.setCurrentMode(Mode.Normal);
```

@feeiyu 
if you have better idea, please tell me..

Reply:
@feeiyu 
now I added withRange parameter to ExecuteNormalCommandTransformation.
implementing it realizes fulfilling start/end line and determining executing normal command with range or not.
I think this is better idea.
[1261062](https://github.com/VSCodeVim/Vim/pull/8896/commits/1261062763a1ab4b48e07a0461b6c65cc6b1d368)

Reply:
Can we replace `startLineNumber`/`endLineNumber`/`withRange` with this?
```ts
range?: LineRange,
```

Reply:
OK, fixed it on 21d94d0